pkg/provider/kubernetes/cluster.go

```diff
@@ -269,13 +269,19 @@ func (p *ClusterProvider) Get(ctx context.Context, userInfo *provider.UserInfo,
 	if options == nil {
 		options = &provider.ClusterGetOptions{}
 	}
-	seedImpersonatedClient, err := createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
-	if err != nil {
-		return nil, err
+
+	client := p.client
+
+	if !userInfo.IsAdmin {
+		var err error
+		client, err = createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
instead of changing the cluster Get, use the GetUnsecured, which is intended for admins

Reply:
Damn, it requires Project object. It's going to be some dependency hell but let me try.

Reply:
Thanks for suggestion @lsviben. IMHO now it looks much better and what's more important doesn't modify the cluster provider.